lib/src/test/java/tech/relaycorp/relaydroid/messaging/SendMessageTest.kt

```diff
@@ -0,0 +1,84 @@
+package tech.relaycorp.relaydroid.messaging
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Assert.assertArrayEquals
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaydroid.PublicThirdPartyEndpoint
+import tech.relaycorp.relaydroid.test.FirstPartyEndpointFactory
+import tech.relaycorp.relaydroid.test.MessageFactory
+import tech.relaycorp.relaynet.bindings.pdc.ServerConnectionException
+import tech.relaycorp.relaynet.messages.Parcel
+import java.time.Duration
+
+internal class SendMessageTest {
+
+    private val poWebClient = mock<PoWebClient>()
+    private val coroutineScope = TestCoroutineScope()
+    private val subject = SendMessage({ poWebClient }, coroutineScope.coroutineContext)
+
+    @Test(expected = InvalidMessageException::class)
+    fun invalidMessage() = coroutineScope.runBlockingTest {
+        val message = OutgoingMessage(
+            ByteArray(0),
+            FirstPartyEndpointFactory.build(),
+            PublicThirdPartyEndpoint("http://example.org")
+        )
+
+        subject.send(message)
+    }
+
+    @Test
+    fun deliverParcelToPublicEndpoint() = coroutineScope.runBlockingTest {
+        val message = MessageFactory.buildOutgoing()
+        subject.send(message)
+
+        verify(poWebClient).deliverParcel(check { parcelSerialized ->
+            val parcel = Parcel.deserialize(parcelSerialized)
+            assertEquals(message.receiverEndpoint.address, parcel.recipientAddress)
+            assertArrayEquals(message.message, parcel.payload)
+            parcel.senderCertificate.let { cert ->
+                cert.validate()
+                assertEquals(message.senderEndpoint.keyPair.public, cert.subjectPublicKey)
+                assertTrue(Duration.between(message.creationDate, cert.startDate).seconds < 2)
+                assertTrue(Duration.between(message.expirationDate, cert.expiryDate).seconds < 2)
+            }
+            assertEquals(message.id.value, parcel.id)
+            assertTrue(Duration.between(message.creationDate, parcel.creationDate).seconds < 2)
+            assertEquals(message.ttl, parcel.ttl)
+            assertArrayEquals(
+                arrayOf(message.senderEndpoint.gatewayCertificate),
+                parcel.senderCertificateChain.toTypedArray()
+            )
```

Comment:
I think these are too many assertions for a single test. Even if we move the building of the parcel to `OutgoingMessage` per my suggestion above, I think the new test should be broken up into smaller, focused ones -- instead of ending up with a single test that checks so many things again.